src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -522,8 +512,14 @@ uct_cuda_copy_mem_alloc(uct_md_h uct_md, size_t *length_p, void **address_p,
     *memh_p    = alloc_handle;
     *address_p = (void*)alloc_handle->ptr;
     *length_p  = alloc_handle->length;
+
 out:
-    uct_cuda_copy_pop_ctx(cu_device, alloc_cu_device, UCS_LOG_LEVEL_ERROR);
+    if (cu_device != alloc_cu_device) {
+        (void)UCT_CUDADRV_FUNC(cuCtxPopCurrent(&ctx), UCS_LOG_LEVEL_WARN);
+        (void)UCT_CUDADRV_FUNC(cuDevicePrimaryCtxRelease(alloc_cu_device),
+                               UCS_LOG_LEVEL_WARN);
```

Comment:
IMO it was better when these 2 calls were in a helper function uct_cuda_copy_pop_ctx because it is the opposite of uct_cuda_copy_push_ctx
then reuse uct_cuda_copy_pop_ctx  also in uct_cuda_copy_md_sync_memops_get_address_range for consistency